pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -167,3 +171,11 @@ func (r *ReconcilePolicy) removeFinalizerAndReleaseMac(request *reconcile.Reques
 
 	return nil
 }
+
+func isVmDeletionAlreadyPersistedByFormerUpdates(err error, parentLogger logr.Logger) bool {
+	if apierrors.IsNotFound(err) {
+		parentLogger.V(1).Info("vm not found")
+		return true
+	}
+	return false
+}
```

Comment:
We only use this function at one place, I see it super over kill `apierror.IsNotFound` is quite clear.

Reply:
yeah this is more for documentation, to explain that the request may be aborted if the vm is already gone (i.e. KMP reconcile already removed the finalizer, but a new vm reconcile request still arrives after the vm is removed)